### DISCUSSION TITLE
How to have published (live) and unpublished (dev) pages, so I can render both live and dev versions of a website?

### DISCUSSION THREAD

**Discussion by beyond2060** [quarto-dev/quarto-cli#9626] [Q&A] [answered]
### Description

What is the best way to have separate development and live versions of a site?

I have a site with pages that are both published and unpublished. When using it locally I want to render all the pages so I see unpublished pages, but when I make it live I want only render the published pages. 

I get the way to do this is to add a custom field to the YAML for a page:

`---`
`title: "My Blog Post"`
`published: false`
`---`

But I can't see any easy way to have render/not render based on this field, other than writing custom code for the render process. It seems to me that there should be a built in way to do this, but I can't find it. Perhaps I'm missing something.

**Comment by beyond2060** [comment]
Ah, just discovered Project Profiles. The documentation doesn't say how to have published and unpublished pages, but I'll try to work that out and close this discussion if I find out the answer.

**Reply by mcanouil** [reply]
Use project profile as documented and set `output-dir` to a different location in each profiles.
Then everything else is the same.

**Reply by beyond2060** [reply]
Thanks Mickeal. How do I specify that I want a page only to be rendered for a particular profile? I can't see that in the documentation.

**Reply by mcanouil** [reply]
As you do normally.
A page inside your "dev" profile configuration file will be rendered only in it.
That's documented, in particular there is a book example.

https://quarto.org/docs/projects/profiles.html

**Reply by beyond2060** [reply]
Ah yes I saw that, but that means you have to list every page you want published in the profile. I guess I could do it that way but it would be better to be able to do it on the pages themselves in the YAML.

**Reply by mcanouil** [reply, marked as answer]
Then you might want to use 1.5 pre-release and the "draft" feature instead: https://quarto.org/docs/prerelease/1.5/website-drafts.html

**Reply by beyond2060** [reply]
That's it thanks!
